<a href="https://colab.research.google.com/github/Vaibhav074N/Assigement-Mar11/blob/main/Assigement_Mar11.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Q1: What is the difference between a t-test and a z-test? Provide an example scenario where you would
use each type of test.

A t-test and a z-test are both statistical tests used to determine whether a sample comes from a population with a known population mean or not. However, the key difference between the two tests is that a z-test is used when the population standard deviation is known, while a t-test is used when the population standard deviation is unknown and estimated from the sample.

An example scenario where a z-test might be used is when a researcher wants to compare the average height of male and female students at a university. If the population standard deviation of heights is known, a z-test can be used to test the hypothesis that the population means of the two groups are equal.

An example scenario where a t-test might be used is when a researcher wants to test the effectiveness of a new drug on reducing cholesterol levels in patients. Since the population standard deviation of cholesterol levels is typically unknown, a t-test can be used to test the hypothesis that the mean cholesterol levels of patients taking the new drug are lower than those not taking the drug

Q2: Differentiate between one-tailed and two-tailed tests.

In hypothesis testing, the significance level is the probability of rejecting the null hypothesis when it is true. A one-tailed test is a hypothesis test in which the null hypothesis is rejected only if the test statistic falls in the critical region in one tail of the distribution. The critical region for a one-tailed test is on one side of the distribution, either to the right or the left.

On the other hand, a two-tailed test is a hypothesis test in which the null hypothesis is rejected if the test statistic falls in either tail of the distribution. The critical region for a two-tailed test is on both sides of the distribution, which means that the null hypothesis is rejected if the test statistic falls in either the right or the left tail of the distribution.

For example, in a one-tailed test, a researcher might hypothesize that a certain drug will increase the average response time of participants on a cognitive task. They would then perform a one-tailed test to determine if the drug does indeed increase the response time. In contrast, in a two-tailed test, a researcher might hypothesize that a certain intervention will change the average score on a test. They would then perform a two-tailed test to determine if the intervention does indeed have an effect on the scores, regardless of whether it increases or decreases them.

Q3: Explain the concept of Type 1 and Type 2 errors in hypothesis testing. Provide an example scenario for
each type of error.

Type 1 error, also known as a false positive, occurs when the null hypothesis is rejected even though it is true. This means that the alternative hypothesis is accepted, even though there is no significant evidence to support it. The probability of making a Type 1 error is denoted by alpha (α), and it is usually set at a significance level of 0.05 or 0.01. An example scenario of Type 1 error is when a medical researcher concludes that a new drug is effective in treating a disease when, in fact, it has no effect.

Type 2 error, also known as a false negative, occurs when the null hypothesis is accepted even though it is false. This means that the alternative hypothesis is rejected, even though there is significant evidence to support it. The probability of making a Type 2 error is denoted by beta (β). An example scenario of Type 2 error is when a medical researcher concludes that a new drug is not effective in treating a disease when, in fact, it is effective.

Q4: Explain Bayes's theorem with an example.

Bayes's theorem is a mathematical formula used in probability theory and statistics to calculate the probability of an event based on prior knowledge of related events. It is named after the Reverend Thomas Bayes, who developed the theorem in the 18th century.

The formula for Bayes's theorem is:

P(A/B) = P(A) * P(B/A) / P(B)

Where,

P(A/B) = probability of A given B is true.

P(B/A) = probability of B given A is true.

P(A) is the Independed probability of event A.

P(B) is the Independed probability of event B.




An example scenario where Bayes's theorem could be used is in medical diagnosis. Suppose a patient is experiencing certain symptoms that could be indicative of a certain disease. The prior probability of the patient having the disease is relatively low, say 5%. The doctor orders a test to confirm the diagnosis. The test has a sensitivity of 90%, which means that if the patient has the disease, the test correctly identifies it 90% of the time. However, the test also has a false positive rate of 10%, which means that if the patient does not have the disease, the test incorrectly indicates that they do 10% of the time.

Using Bayes's theorem, we can calculate the probability that the patient actually has the disease given that the test is positive. Let's assume the test comes back positive. We can calculate the probability of the patient having the disease given the positive test result as follows:

P(disease|positive test) = P(positive test|disease) x P(disease) / P(positive test)

where:
P(disease|positive test) is the probability of the patient having the disease given a positive test result.
P(positive test|disease) is the probability of a positive test result given that the patient has the disease, which is the sensitivity of the test (0.90).
P(disease) is the prior probability of the patient having the disease (0.05).
P(positive test) is the overall probability of a positive test result, which is the sum of the probability of a true positive (0.90 x 0.05) and the probability of a false positive (0.10 x 0.95) = 0.135.

Plugging in the numbers, we get:

P(disease|positive test) = 0.90 x 0.05 / 0.135 = 0.333

So the probability of the patient actually having the disease given a positive test result is 33.3%. This shows that even with a relatively high sensitivity, the probability of a positive test result accurately indicating the presence of a disease can still be relatively low, especially if the prior probability of the disease is low.

Q5: What is a confidence interval? How to calculate the confidence interval, explain with an example.

A confidence interval is a range of values that is likely to contain the true value of a population parameter with a certain level of confidence. It is used to estimate the true value of a population parameter, such as the mean or proportion, based on a sample of data.

To calculate a confidence interval, we need to first determine the sample mean and sample standard deviation, as well as the sample size and the desired level of confidence. The formula for a confidence interval for a population mean is:

Confidence interval = sample mean ± (critical value) x (standard error)

where the critical value is determined by the level of confidence and the degrees of freedom, and the standard error is calculated as the standard deviation divided by the square root of the sample size

In [ ]:
import numpy as np
from scipy.stats import t

n = 50
sample_mean = 10
sample_std = 2
df = n - 1

t_value = t.ppf(0.975, df)

# Margin of error
moe = t_value * (sample_std / np.sqrt(n))

lower_bound = sample_mean - moe
upper_bound = sample_mean + moe

print(f"95% confidence interval: ({lower_bound}, {upper_bound})")

95% confidence interval: (9.431606289755253, 10.568393710244747)


Q6. Use Bayes' Theorem to calculate the probability of an event occurring given prior knowledge of the
event's probability and new evidence. Provide a sample problem and solution.

Problem: A medical test is known to correctly identify 95% of cases of a certain disease, but also has a false positive rate of 10%. The disease is known to occur in 1% of the population. If a person tests positive for the disease, what is the probability that they actually have the disease?

Solution:

Let A be the event that the person has the disease, and B be the event that the person tests positive for the disease. Then, we want to find P(A|B), the probability of having the disease given a positive test result.

We can use Bayes' Theorem to calculate this probability:

P(A|B) = P(B|A) * P(A) / P(B)

where:

P(B|A) is the probability of testing positive given that the person has the disease, which is the sensitivity of the test, or 0.95 in this case.
P(A) is the prior probability of having the disease, which is 0.01.
P(B) is the total probability of testing positive, which can be calculated using the law of total probability:
P(B) = P(B|A) * P(A) + P(B|not A) * P(not A)
where P(B|not A) is the false positive rate, or 0.1 in this case, and P(not A) is 1 - P(A), or 0.99 in this case.
Substituting these values into Bayes' Theorem, we get:

P(A|B) = 0.95 * 0.01 / (0.95 * 0.01 + 0.1 * 0.99) = 0.086

Therefore, the probability of actually having the disease given a positive test result is only 8.6%. This shows the importance of considering both the sensitivity and false positive rate of a test, as well as the prior probability of the event, when interpreting test results

Q7. Calculate the 95% confidence interval for a sample of data with a mean of 50 and a standard deviation
of 5. Interpret the results.

In [1]:
import scipy.stats as stats

mean = 50
std_dev = 5
n = 100  

std_err = std_dev / (n ** 0.5)

t_value = stats.t.ppf(0.975, n-1)

lower_ci = mean - t_value * std_err
upper_ci = mean + t_value * std_err

print("95% Confidence Interval: ({:.2f}, {:.2f})".format(lower_ci, upper_ci))

95% Confidence Interval: (49.01, 50.99)


Q8. What is the margin of error in a confidence interval? How does sample size affect the margin of error?
Provide an example of a scenario where a larger sample size would result in a smaller margin of error.

The margin of error is the range of values above and below the sample statistic (such as the sample mean or proportion) that is likely to contain the true population parameter with a certain level of confidence. It represents the uncertainty in the estimation of the population parameter based on the sample.

The formula for the margin of error is:

Margin of error = z-score x (standard deviation / square root of sample size)

where the z-score is determined by the desired confidence level.

Sample size affects the margin of error in two ways: as the sample size increases, the margin of error decreases, and as the sample size decreases, the margin of error increases. This is because a larger sample size provides more information about the population, leading to a more accurate estimation of the population parameter and less uncertainty

In [5]:
import numpy as np
import scipy.stats as stats

data1 = np.random.normal(loc=50, scale=5, size=100)
data2 = np.random.normal(loc=50, scale=5, size=500)

mean1, std1 = np.mean(data1), np.std(data1, ddof=1)
mean2, std2 = np.mean(data2), np.std(data2, ddof=1)

z_score = stats.norm.ppf(0.975)
moe1 = z_score * std1 / np.sqrt(len(data1))
moe2 = z_score * std2 / np.sqrt(len(data2))

In [4]:
z_score

1.959963984540054

In [6]:
print("Margin of error for sample size 100: {:.2f}".format(moe1))
print("Margin of error for sample size 500: {:.2f}".format(moe2))

Margin of error for sample size 100: 0.92
Margin of error for sample size 500: 0.45


Q9. Calculate the z-score for a data point with a value of 75, a population mean of 70, and a population
standard deviation of 5. Interpret the results.

In [7]:
xbar = 75
mean = 70
stddev = 5

z_score = (xbar - mean) / stddev
print("The z-score is:", z_score)

The z-score is: 1.0


Q10. In a study of the effectiveness of a new weight loss drug, a sample of 50 participants lost an average
of 6 pounds with a standard deviation of 2.5 pounds. Conduct a hypothesis test to determine if the drug is
significantly effective at a 95% confidence level using a t-test.

Ans:

Null hypothesis: The weight loss drug is not significantly effective.

Alternative hypothesis: The weight loss drug is significantly effective

In [8]:
import scipy.stats as stats

# sample statistics
n = 50
x_bar = 6
s = 2.5

# hypothesized population parameters
mu = 0

# calculate the t-statistic
t = (x_bar - mu) / (s / (n ** 0.5))

# calculate the critical t-value
alpha = 0.05
df = n - 1
t_crit = stats.t.ppf(alpha/2, df)

# calculate the p-value
p_value = stats.t.sf(abs(t), df) * 2

# print the results
print("t-statistic:", t)
print("critical t-value:", -t_crit, t_crit)
print("p-value:", p_value)

t-statistic: 16.970562748477143
critical t-value: 2.0095752344892093 -2.0095752344892093
p-value: 3.7168840835270203e-22


The calculated t-statistic of 12.00 is greater than the critical t-value of ±2.0096, and the p-value is less than 0.05. Therefore, we reject the null hypothesis and conclude that the new weight loss drug is significantly effective at a 95% confidence level

Q11. In a survey of 500 people, 65% reported being satisfied with their current job. Calculate the 95%
confidence interval for the true proportion of people who are satisfied with their job.

To calculate the 95% confidence interval for the proportion of people satisfied with their job, we can use the formula:

CI = p ± zsqrt((p(1-p))/n)

where:

p is the proportion of people satisfied with their job in the sample
z is the critical value from the standard normal distribution for a 95% confidence level (which is approximately 1.96)
n is the sample size
Plugging in the values from the problem, we get:

p = 0.65

z = 1.96

n = 500

CI = 0.65 ± 1.96sqrt((0.65(1-0.65))/500)

In [9]:
import math

p = 0.65
n = 500
z = 1.96 # 95% confidence level
margin_of_error = z * math.sqrt((p*(1-p))/n)

lower_bound = p - margin_of_error
upper_bound = p + margin_of_error

print(lower_bound)
print(upper_bound)

0.608191771144905
0.6918082288550951


Q12. A researcher is testing the effectiveness of two different teaching methods on student performance.
Sample A has a mean score of 85 with a standard deviation of 6, while sample B has a mean score of 82
with a standard deviation of 5. Conduct a hypothesis test to determine if the two teaching methods have a
significant difference in student performance using a t-test with a significance level of 0.01.

Step 1: State the null and alternative hypotheses:
Null hypothesis: The mean scores of the two teaching methods are equal (μA = μB).
Alternative hypothesis: The mean scores of the two teaching methods are not equal (μA ≠ μB).

Step 2: Set the significance level:
Significance level (α) = 0.01

Step 3: Determine the test statistic:
We will use a two-sample t-test to compare the means of the two groups. The test statistic can be calculated as:
t = (x̄A - x̄B) / sqrt((s^2A/nA) + (s^2B/nB))
where x̄A and x̄B are the sample means, s^2A and s^2B are the sample variances, and nA and nB are the sample sizes.

Step 4: Determine the degrees of freedom:

Step 5: Determine the critical value:

Step 6: Make a decision and interpret the results

In [13]:
import numpy as np
from scipy.stats import t

# Sample A
n_a = 30
mean_a = 85
std_a = 6

# Sample B
n_b = 25
mean_b = 82
std_b = 5

df = n_a + n_b - 2

se = np.sqrt((std_a**2/n_a) + (std_b**2/n_b))

t_stat = (mean_a - mean_b) / se

p_value = 2 * (1 - t.cdf(abs(t_stat), df))

alpha = 0.01

In [12]:
p_value , alpha

(0.04817304403676115, 0.01)

In [14]:
if p_value < alpha:
    print("There is a significant difference in student performance between the two teaching methods.")
else:
    print("There is no significant difference in student performance between the two teaching methods.")

There is no significant difference in student performance between the two teaching methods.


Q13. A population has a mean of 60 and a standard deviation of 8. A sample of 50 observations has a mean
of 65. Calculate the 90% confidence interval for the true population mean.

To calculate the 90% confidence interval for the true population mean, we can use the formula:

CI = X̄ ± z*(σ/√n)

where X̄ is the sample mean, σ is the population standard deviation, n is the sample size, and z is the critical value from the standard normal distribution for the desired level of confidence.

First, we need to find the critical value for a 90% confidence level. Using a standard normal distribution table or calculator, we find that the critical value is 1.645.

Plugging in the values, we get:

CI = 65 ± 1.645*(8/√50)
= 65 ± 2.31

Therefore, the 90% confidence interval for the true population mean is (63.13, 66.86). This means we can be 90% confident that the true population mean falls within this range

In [19]:
import math

sample_mean = 65
pop_mean = 60
pop_stddev = 8
sample_size = 50
confidence_level = 0.90

std_error = pop_stddev / math.sqrt(sample_size)

z_score = abs(stats.norm.ppf((1 - confidence_level) / 2))

margin_of_error = z_score * std_error

lower_bound = sample_mean - margin_of_error
upper_bound = sample_mean + margin_of_error

In [20]:
lower_bound , upper_bound

(63.13906055411732, 66.86093944588268)

Q14. In a study of the effects of caffeine on reaction time, a sample of 30 participants had an average
reaction time of 0.25 seconds with a standard deviation of 0.05 seconds. Conduct a hypothesis test to
determine if the caffeine has a significant effect on reaction time at a 90% confidence level using a t-test.

In [22]:
import numpy as np
from scipy.stats import t

n = 30
xbar = 0.25
s = 0.05
mu0 = 0.24

t_stat = (xbar - mu0) / (s / np.sqrt(n))

alpha = 0.1
df = n - 1
t_crit = t.ppf(1 - alpha/2, df)

In [23]:
t_stat

1.0954451150103333

In [24]:
t_crit

1.6991270265334972

In [ ]:
if abs(t_stat) > t_crit:
    print("Reject the null hypothesis - caffeine has a significant effect on reaction time")
else:
    print("Fail to reject the null hypothesis - there is not enough evidence to conclude that caffeine has a significant effect on reaction time")